src/java/org/apache/cassandra/db/virtual/AbstractSchemaMetadataTable.java

```
@@ -0,0 +1,299 @@
+                           .addPartitionKeyColumn(KEYSPACE_NAME, UTF8Type.instance)
+                           .addClusteringColumn(TABLE_NAME, UTF8Type.instance)
+                           .addClusteringColumn(COLUMN_NAME, UTF8Type.instance)
+                           .addClusteringColumn(UDT_NAME, UTF8Type.instance)
+                           .addRegularColumn(schemaTableType.columnName, UTF8Type.instance)
+                           .build());
+        this.schemaTableType = schemaTableType;
+    }
+
+    /**
+     * Extract metadata from a keyspace.
+     * @param keyspace the keyspace metadata
+     * @return the metadata value, or null if not present
+     */
+    protected abstract String extractKeyspaceMetadata(KeyspaceMetadata keyspace);
+
+    /**
+     * Extract metadata from a table.
+     * @param table the table metadata
+     * @return the metadata value, or null if not present
+     */
+    protected abstract String extractTableMetadata(TableMetadata table);
+
+    /**
+     * Extract metadata from a column.
+     * @param column the column metadata
+     * @return the metadata value, or null if not present
+     */
+    protected abstract String extractColumnMetadata(ColumnMetadata column);
+
+    /**
+     * Extract metadata from a UDT.
+     * @param udt the user-defined type
+     * @return the metadata value, or null if not present
+     */
+    protected abstract String extractUdtMetadata(UserType udt);
+
+    @Override
+    public DataSet data()
+    {
+        SimpleDataSet result = new SimpleDataSet(metadata());
+        SchemaProvider schemaProvider = Schema.instance;
+        for (String keyspaceName : schemaProvider.getKeyspaces())
+        {
+            KeyspaceMetadata keyspace = schemaProvider.getKeyspaceMetadata(keyspaceName);
+            if(keyspace == null)
+            {
+                noSpamLogger.warn("Keyspace metadata not found for keyspace: {}", keyspaceName);
```

I am not completely sure we want to see this. What is that good for?

---

This is a warning on the server side which could be used for debugging. As per our discussion, I have updated the query to fail instead of returning zero rows if the keyspace name or object type are incorrect.

---

It is very unusual to log like this, I would remove this altogether.